saleor/checkout/utils.py

```diff
@@ -783,6 +784,8 @@ def add_voucher_to_checkout(
             checkout=checkout_info.checkout,
             type=DiscountType.ORDER_PROMOTION,
         ).delete()
+        # delete gift line if exists
```

Comment:
You're operating on `lines` build in `checkoutAddPromoCode`, lines could be changed by another mutation. So you could currently operate on not up-to-date lines. So potentially we could have a voucher and gift line assigned 🤔  correct?